crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,400 @@
+    // Test 4 owners with threshold 2 - only 2 signatures should be needed
+
+    // Setup keys and authenticators (4 approvers, but only 2 signers)
+    let (_secret_keys, public_keys, authenticators) =
+        setup_keys_and_authenticators(4, 2, [1u8; 32])?;
+
+    // Create multisig account with 4 approvers but threshold of 2
+    let multisig_account = create_multisig_account(2, &public_keys, [1; 32], 10)?;
+
+    let mut mock_chain = MockChainBuilder::with_accounts([multisig_account.clone()])
+        .unwrap()
+        .build()
+        .unwrap();
+
+    mock_chain.prove_next_block()?;
+
+    // Create output note for the transaction
+    let output_note = create_p2id_note(
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_IMMUTABLE_CODE.try_into().unwrap(),
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE.try_into().unwrap(),
+        vec![],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new(Word::from([4u32; 4])),
+    )?;
+
+    let asset = FungibleAsset::mock(5);
+    let tx_script = TransactionScript::compile(
+        format!(
+            "
+            use.miden::tx
+            begin
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.0              # aux
+                push.{tag}
+                call.tx::create_note
+
+                push.{asset}
+                call.::miden::contracts::wallets::basic::move_asset_to_note
+                dropw dropw dropw dropw
+            end
+            ",
+            recipient = word_to_masm_push_string(&output_note.recipient().digest()),
+            note_type = NoteType::Public as u8,
+            tag = Felt::from(output_note.metadata().tag()),
+            asset = word_to_masm_push_string(&asset.into()),
+            note_execution_hint = Felt::from(output_note.metadata().execution_hint()),
+        ),
+        TransactionKernel::testing_assembler(),
+    )?;
+
+    let salt = Word::from([Felt::new(2); 4]);
+
+    // Execute transaction without signatures - should fail
+    let tx_context_init = mock_chain
+        .build_tx_context(multisig_account.id(), &[], &[])?
+        .tx_script(tx_script.clone())
+        .extend_expected_output_notes(vec![OutputNote::Full(output_note.clone())])
+        .auth_args(salt)
+        .build()?;
+
+    let tx_summary = match tx_context_init.execute().unwrap_err() {
+        TransactionExecutorError::Unauthorized(tx_effects) => tx_effects,
+        error => panic!("expected abort with tx effects: {error:?}"),
+    };
+
+    // Get signatures from only 2 of the 4 approvers (should be sufficient for threshold 2)
+    let msg = tx_summary.as_ref().to_commitment();
+    let tx_summary = SigningInputs::TransactionSummary(tx_summary);
+
+    let sig_1 = authenticators[0].get_signature(public_keys[0].into(), &tx_summary)?;
+    let sig_2 = authenticators[1].get_signature(public_keys[1].into(), &tx_summary)?;
```

Comment:
I would also add a test where the signers are not necessarily the first in the approvers list

Reply:
Potentially this test could be refactored to test more (or even better: all possible) combinations of the 2/4 approvers?